src/rpc/methods/state.rs

```diff
@@ -2800,3 +2800,84 @@ impl TryFrom<&ChainConfig> for ForkUpgradeParams {
         })
     }
 }
+
+pub enum StateMinerInitialPledgeForSector {}
+
+impl RpcMethod<4> for StateMinerInitialPledgeForSector {
+    const NAME: &'static str = "Filecoin.StateMinerInitialPledgeForSector";
+    const PARAM_NAMES: [&'static str; 4] = [
+        "sector_duration",
+        "sector_size",
+        "verified_size",
+        "tipset_key",
+    ];
+    const API_PATHS: ApiPaths = ApiPaths::V1;
+    const PERMISSION: Permission = Permission::Read;
+
+    type Params = (ChainEpoch, SectorSize, u64, ApiTipsetKey);
+    type Ok = TokenAmount;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (sector_duration, sector_size, verified_size, ApiTipsetKey(tsk)): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        if sector_duration <= 0 {
+            return Err(anyhow::anyhow!("sector duration must be greater than 0").into());
+        }
+        let sec_size: u64 = match sector_size {
+            SectorSize::_2KiB => 2 * 1024,
+            SectorSize::_8MiB => 8 * 1024 * 1024,
+            SectorSize::_512MiB => 512 * 1024 * 1024,
+            SectorSize::_32GiB => 32 * 1024 * 1024 * 1024,
+            SectorSize::_64GiB => 64 * 1024 * 1024 * 1024,
+        };
+
+        if sec_size == 0 {
+            return Err(anyhow::anyhow!("sector size must be non-zero").into());
+        }
+
+        if sec_size < verified_size {
+            return Err(
+                anyhow::anyhow!("verified deal size cannot be larger than sector size").into(),
+            );
+        }
+
+        let ts = ctx.chain_store().load_required_tipset_or_heaviest(&tsk)?;
+
+        let power_state: power::State = ctx.state_manager.get_actor_state(&ts)?;
+        let power_smoothed = power_state.total_power_smoothed();
+        let pledge_collateral = power_state.total_locked();
+
+        let reward_state: reward::State = ctx.state_manager.get_actor_state(&ts)?;
+
+        let genesis_info = GenesisInfo::from_chain_config(ctx.chain_config().clone());
+        let circ_supply = genesis_info.get_vm_circulating_supply_detailed(
+            ts.epoch(),
+            &ctx.store_owned(),
+            ts.parent_state(),
+        )?;
+
+        let deal_weight = BigInt::from(0);
+        let verified_deal_weight = BigInt::from(verified_size) * sector_duration;
+        let sector_weight = qa_power_for_weight(
+            sector_size.into(),
+            sector_duration,
+            &deal_weight,
+            &verified_deal_weight,
+        );
+
+        let initial_pledge: TokenAmount = reward_state
+            .initial_pledge_for_power(
+                &sector_weight,
+                pledge_collateral,
+                power_smoothed,
+                &circ_supply.fil_circulating.into(),
+                power_state.ramp_start_epoch(),
```

Comment:
Lotus is using some ramp but not us. Please see.

https://github.com/filecoin-project/lotus/blob/a5847457fd2e143d643a1a8326797e4402d2c527/node/impl/full/state.go#L1588

Reply:
Please carefully review the Lotus method implementation at line 1558. I don't think we are calculating those variables the same way:

`epochsSinceRampStart, rampDurationEpochs, err := a.getPledgeRampParams(ctx, ts.Height(), state)`

Lotus is calling some `getPledgeRampParams` function.